state/cleanup.go

```diff
@@ -212,7 +218,41 @@ func (st *State) cleanupMachinesForDyingModel() (err error) {
 	return nil
 }
 
-// cleanupServicesForDyingModel sets all services to Dying, if they are
+// cleanupVolumesForDyingModel sets all persistent volumes to Dying,
+// if they are not already Dying or Dead. It's expected to be used when
+// a model is destroyed.
+func (st *State) cleanupVolumesForDyingModel() (err error) {
+	volumes, err := st.AllVolumes()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	for _, v := range volumes {
+		err := st.DestroyVolume(v.VolumeTag())
+		if err != nil && !IsContainsFilesystem(err) {
+			return errors.Trace(err)
```

Comment:
I wonder if this error should be typed so that we can avoid log noise. Wouldn't it be the case that the filesystem cleanup would take care of the underlying volume? So a dying model should see both filesystems and volumes destroyed as expected.